### PR TITLE
#1104 Index fixes

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes duplicate indexes that have accumulated due to a dodgy UNIQUE constraint.

I also made a few tweaks to existing migrations to minimise reported errors.

(You should be able to run `yarn migrate 0.0.0`) and not have anything break.

### CODE REVIEW

File: database/buildSchema/43_views_functions_triggers.sql [hunk 1]
```diff
@@ -162,10 +162,6 @@ CREATE TRIGGER trigger_queue
 
 DROP FUNCTION IF EXISTS public.set_template_verision CASCADE; -- DEPRECATED
 
--- Enforce unique versionId per template type
-ALTER TABLE public.template
-    ADD UNIQUE (code, version_id);
-
```
Comment (CarlosNZ) [inline database/buildSchema/43_views_functions_triggers.sql:168]: This is what caused the duplicate template indexes. Doesn't need to be in here, just added the corrected version of this to the migration script instead.

File: documentation/API.md [hunk 2]
```diff
@@ -309,7 +309,7 @@ Returns (on success):
 
 GET: `/check-trigger?serial=<applicationSerial>`
 
-This endpoint is used by the front-end application loader to ensure that ALL triggers associated with an application (i.e. review assignments, reviews, verifications) are not runninng before the application data is fetched. This is so any mutations that cause Actions to run are all finished before any subsequent data is re-fetched, so the front-end shows all the changes.
+This endpoint is used by the front-end application loader to ensure that ALL triggers associated with an application (i.e. review assignments, reviews, verifications) are not running before the application data is fetched. This is so any mutations that cause Actions to run are all finished before any subsequent data is re-fetched, so the front-end shows all the changes.
```
Comment (CarlosNZ) [inline documentation/API.md:312]: Random typo fix 😄

**Comment by CarlosNZ** [comment]
Hmm, @andreievg I just noticed this is based of your branch which is now closed. Might have to be careful when rebasing in case this has pulled in changes we don't want.

**Comment by CarlosNZ** [comment]
> Hmm, @andreievg I just noticed this is based of your branch which is now closed. Might have to be careful when rebasing in case this has pulled in changes we don't want.

Okay @andreievg , I've rebased this off my earlier PR now, and reverted all the changes from your now-obsolete PR.

**Comment by andreievg** [comment]
> Okay @andreievg , I've rebased this off my earlier PR now, and reverted all the changes from your now-obsolete PR.

Thanks @CarlosNZ. 
Sorry I thought that my changes were independent, didn't realise it was a base, otherwise would have made the changes in that base branch. Maybe we can do another branch for build and merge all the pending PRs to it, might help in avoiding these sorts of conflicts

File: database/migration/databaseMethods.ts [hunk 3]
```diff
@@ -255,64 +255,6 @@ const databaseMethods = {
       }
     }
   },
-  updatePermissionPolicyRules: async () => {
```
Comment (andreievg) [inline database/migration/databaseMethods.ts:258]: thanks, i forgot to remove this in my latest version

File: database/migration/migrateData.ts [hunk 4]
```diff
@@ -1135,6 +1147,10 @@ const migrateData = async () => {
   )
   console.log(' - Updating indexes...')
   await DB.changeSchema(createIndexesScript)
+  // This shouldn't be necessary after v1.0 migration, but can't hurt in case
```
Comment (andreievg) [inline database/migration/migrateData.ts:1150]: 🙏

**Pull request merged**